### PR TITLE
Make space ID optional on storage reports

### PR DESCRIPTION
**Pull request by max-zilla**
This makes a few minor changes to make the /api/reports/storage/spaces endpoint not require a space ID but run the report on whole database if none is given.

### CODE REVIEW

**Review by bodom0015 — CHANGES_REQUESTED** [review]
Swagger may be slightly incorrect for passing the space id? Your `swagger.yaml` changes say that `space` is supposed to be provided in the path, but I can't seem to figure out how to pass a space id to this endpoint.

I've tried passing it in the path (`/api/reports/storage/spaces/5f7647058d6a63c38911fc78`), but I think we either need to add `{space}` to the path or change the value of `in` for this parameter. This is the error I get passing the space id via that path:
```json
{"status":"not found","request":"GET /api/reports/storage/spaces/5f7647058d6a63c38911fc78"}
```

I also tried `/api/reports/storage/spaces&space=5f7647058d6a63c38911fc78`, but that yielded the same error:
```json
{"status":"not found","request":"GET /api/reports/storage/spaces&space=5f7647058d6a63c38911fc78"}
```

**Comment by max-zilla** [comment]
> `/api/reports/storage/spaces` seems to work, but Swagger may be slightly incorrect for passing the space id? Your `swagger.yaml` changes say that `space` is supposed to be provided in the path, but I can't seem to figure out how to pass a space id to this endpoint. What is the expected URL/format for passing a space id here?
> 
> I've tried passing it in the path (`/api/reports/storage/spaces/5f7647058d6a63c38911fc78`), but I think we either need to add `{space}` to the path or change the `in: path` for this parameter to something else. This is the error I get passing the space id via that path:
> 
> ```json
> {"status":"not found","request":"GET /api/reports/storage/spaces/5f7647058d6a63c38911fc78"}
> ```
> 
> I also tried `/api/reports/storage/spaces&space=5f7647058d6a63c38911fc78`, but that yielded the same error:
> 
> ```json
> {"status":"not found","request":"GET /api/reports/storage/spaces&space=5f7647058d6a63c38911fc78"}
> ```

The intended syntax was your second one, but I changed & to ?
`/api/reports/storage/spaces?space=5f7647058d6a63c38911fc78`
...let me double-check syntax + docs, thx.

**Comment by bodom0015** [comment]
🤦  you are correct! `/api/reports/storage/spaces?space=5f7647058d6a63c38911fc78` worked like a charm

When using the proper syntax, the endpoint works as expected :+1:

File: public/swagger.yml [hunk 1]
```diff
@@ -4995,18 +4995,18 @@ paths:
...
         - reporting
       summary: Download storage metrics report for spaces as CSV.
       description: |
-        Download storage metrics report for files in a space as a CSV file.
+        Download storage metrics report for files in a space as a CSV file. Space id is optional parameter.
         Must be a server admin to access the report. Supports ISO8601 date range filters.
       parameters:
-        - name: id
+        - name: space
           in: path
```
Comment (bodom0015) [inline public/swagger.yml:5008]: I think this is the expected syntax for denoting query-string parameters:
```suggestion
          in: query
```

Comment (lmarini) [reply public/swagger.yml]: fixed, thank you @bodom0015

**Review by lmarini — APPROVED** [review]

File: public/swagger.yml (hunk 1 above)
Comment (max-zilla) [reply public/swagger.yml]: thank you

**Pull request merged**